allennlp/confidence_checks/task_checklists/__init__.py

```diff
@@ -1,20 +1,10 @@
-import warnings
-
-try:
-    from allennlp.confidence_checks.task_checklists.task_suite import TaskSuite
-    from allennlp.confidence_checks.task_checklists.sentiment_analysis_suite import (
-        SentimentAnalysisSuite,
-    )
-    from allennlp.confidence_checks.task_checklists.question_answering_suite import (
-        QuestionAnsweringSuite,
-    )
-    from allennlp.confidence_checks.task_checklists.textual_entailment_suite import (
-        TextualEntailmentSuite,
-    )
-except ImportError:
-    warnings.warn(
-        "The checklist integration of allennlp is optional; if you're using conda, "
-        "it can be installed with `conda install allennlp-checklist`, "
-        "otherwise use `pip install allennlp[checklist]`."
-    )
-    raise
```

Comment:
I thought about keeping this warning, but I find the situation somewhat inconsistent. People can similarly do direct imports of anything beyond `task_checklists`, e.g.
``` 
import allennlp.confidence_checks.task_checklists.textual_entailment_suite
```
which has no such guards for the absence of `checklist`. If those modules were private (and only coming through `task_checklists`, the situation would be different IMO).